File: frameworks/go/fiber/Implementation/contract.go
package implementation

import (
	"runtime"
	"runtime/debug"
	"strings"

	"github.com/gofiber/fiber/v3"
)

// contractRoutes answers /health and /__meta, which the contract asks of every framework
// outside the corpus.
func contractRoutes(app *fiber.App) {
	// The payloads are loaded before the server listens, so a server that answers has them.
	app.Get("/health", func(c fiber.Ctx) error { return c.SendString("ok") })

	version := moduleVersion("github.com/gofiber/fiber/v3")
	app.Get("/__meta", func(c fiber.Ctx) error {
		return c.JSON(fiber.Map{
			"framework": "Fiber",
			"version":   version,
			"runtime":   runtime.Version(),
			"adapter":   "fasthttp",
			// Fiber's JSONEncoder, which c.JSON calls, is encoding/json's Marshal unless the config names another.
			"serializer": "encoding/json",
			// Go sets this from the container's CPU quota, so it says how many threads run Go code.
			"gomaxprocs": runtime.GOMAXPROCS(0),
		})
	})
}

// moduleVersion is the version of a module the build resolved, without the v Go puts in front of it.
func moduleVersion(path string) string {
	if info, ok := debug.ReadBuildInfo(); ok {
		for _, dep := range info.Deps {
			if dep.Path == path {
				return strings.TrimPrefix(dep.Version, "v")
			}
		}
	}
	return "unknown"
}

File: frameworks/rust/salvo/Implementation/routes/contract.rs
use salvo::prelude::*;
use serde::Serialize;

/// /__meta: what ran, recorded on the result row and never checked. build.rs reads the versions
/// from Cargo.lock and rustc when the binary is built.
#[derive(Serialize)]
struct Meta {
    framework: &'static str,
    version: &'static str,
    runtime: &'static str,
    serializer: &'static str,
}

const META: Meta = Meta {
    framework: "salvo",
    version: env!("RB_SALVO_VERSION"),
    runtime: env!("RB_RUSTC_VERSION"),
    serializer: concat!("serde_json ", env!("RB_SERDE_JSON_VERSION")),
};

/// The payloads are loaded before the server starts, so a server that answers has them.
#[handler]
async fn health() -> &'static str {
    "ok"
}

#[handler]
async fn meta() -> Json<Meta> {
    Json(META)
}

/// /health and /__meta, which the contract asks of every framework outside the corpus.
pub fn router() -> Router {
    Router::new()
        .push(Router::with_path("/health").get(health))
        .push(Router::with_path("/__meta").get(meta))
}
